packages/core/modules-sdk/src/remote-link.ts

```diff
@@ -453,4 +455,72 @@ export class RemoteLink {
   ): Promise<[CascadeError[] | null, RestoredIds]> {
     return await this.executeCascade(removedServices, "restore")
   }
+
+  async list(
+    link: LinkDefinition | LinkDefinition[],
+    options?: { asLinkDefinition?: boolean }
+  ): Promise<(object | LinkDefinition)[]> {
+    const allLinks = Array.isArray(link) ? link : [link]
+    // TODO: create a generic function that builds a serviceLinkMap based on data
+    const serviceLinks = new Map<string, object[]>()
+
+    for (const rel of allLinks) {
+      const mods = Object.keys(rel).filter((mod) => mod !== "data")
+
+      if (mods.length > 2) {
+        throw new Error(`Only two modules can be linked.`)
+      }
+
+      const [moduleA, moduleB] = mods
+      const pk = Object.keys(rel[moduleA])
+      const moduleAKey = pk.join(",")
+      const moduleBKey = Object.keys(rel[moduleB]).join(",")
+
+      const service = this.getLinkModule(
+        moduleA,
+        moduleAKey,
+        moduleB,
+        moduleBKey
+      )
+
+      if (!service) {
+        throw new Error(
+          linkingErrorMessage({
+            moduleA,
+            moduleAKey,
+            moduleB,
+            moduleBKey,
+            type: "link",
+          })
+        )
+      } else if (!serviceLinks.has(service.__definition.key)) {
+        serviceLinks.set(service.__definition.key, [])
+      }
+
+      serviceLinks.get(service.__definition.key)?.push({
+        ...rel[moduleA],
+        ...rel[moduleB],
+      })
+    }
+
+    const promises: Promise<object[]>[] = []
+
+    for (const [serviceName, filters] of serviceLinks) {
+      const service = this.modulesMap.get(serviceName)!
+
+      promises.push(
+        service
+          .list({ $or: filters })
+          .then((links: any[]) =>
+            options?.asLinkDefinition
+              ? convertRecordsToLinkDefinition(links, service)
```

Comment:
@carlos-r-l-rodrigues please take a look at this method, more specifically this transformation.

Outside of this class, its hard for workflows to accurately perform a revert because it doesn't have information about which fields in the database are the primary keys or which one are extra "data" fields.

I've added a helper in this class to convert database records to link definition shape, which the workflows can then pass around to all the create/update/dismiss methods. I quite liked using it as it made the workflows super easy. 

wdyt? Let me know if you have a better idea 🤞🏻

Reply:
This looks good. Nice addition to it.